pkg/cfn/manager/delete_tasks.go

```diff
@@ -115,6 +115,63 @@ func (c *StackCollection) NewTasksToDeleteNodeGroups(shouldDelete func(string) b
 	return taskTree, nil
 }
 
+type DeleteWaitCondition struct {
+	Condition func() (bool, error)
+	Timeout   time.Duration
+	Interval  time.Duration
+}
+
+type DeleteUnownedNodegroupTask struct {
+	cluster   string
+	nodegroup string
+	wait      *DeleteWaitCondition
+	info      string
+	eksAPI    eksiface.EKSAPI
+}
+
+func (d *DeleteUnownedNodegroupTask) Describe() string {
+	return d.info
+}
+
+func (d *DeleteUnownedNodegroupTask) Do(errorchan chan error) error {
+	out, err := d.eksAPI.DeleteNodegroup(&eks.DeleteNodegroupInput{
+		ClusterName:   &d.cluster,
+		NodegroupName: &d.nodegroup,
+	})
+	go func() {
```

Comment:
I think you can use https://github.com/weaveworks/eksctl/blob/8b2a568175167a53bfe09a700b68ffa4f65a0162/pkg/eks/nodegroup.go#L361 here?